requirements-convert-lora-to-ggml.txt

```diff
@@ -0,0 +1,2 @@
+-r requirements-convert.txt
+torch==2.1.1
```

Comment:
Pinning to a specific patch release is not a good idea, although the existing requirements files also have this problem. `torch~=2.1.1` will allow the current torch bugfix release, `2.1.2`, to be used as well.

Reply:
I updated all the existing '==' to '~=' and also added a (suppressible if needed) automated check to prevent those from being added unintentionally.

When looking up the requirements.txt spec, it looks like it's possible to specify major versions like `torch>=2.1.0, torch<3.0.0`. Would that be preferable or is that loose of a constraint going to cause problems? In dart land, that's a fairly typical package dependency, but maybe dart ecosystem follows semver more closely than typical python.

Reply:
It's a case-by-case thing. torch hasn't always adhered to semver; I've known its numerical output to change between minor releases. sentencepiece is still 0.1.x.

transformers and protobuf would be good candidates for allowing future minor versions, and they need to be fixed anyway  - `transformers>=4.35.2,<5.0.0` should be used in both places, and protobuf should be specified as `protobuf>=4.21.0,<5.0.0`.